### ISSUE TITLE
cube and shadow have validation errors

### ISSUE THREAD

**Issue by m4b** [gfx-rs/wgpu#132]
I'm seeing validation errors for the cube and shadow `examples-gfx`:

```
m4b@efrit ::  [ ~/git/wgpu ] target/debug/cube 
Xlib:  extension "NV-GLX" missing on display ":0".
ERROR 2019-04-25T06:29:06Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x1f (Type = 8) | Mapping Memory from 0x0 to 0x80 oversteps total array size 0x48. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
```

os: archlinux
hash: 0a60c82c435e2d570270cd348803d989ccf31929
some vulkaninfo:
```
Vulkan Instance Version: 1.1.96
=====================
GPU id       : 0 (Intel(R) HD Graphics 620 (Kaby Lake GT2))
```

Here is shadow validation errors (appear to be same/similar):
```
ERROR 2019-04-25T06:28:56Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x10fe (Type = 8) | Mapping Memory from 0x0 to 0x1c0 oversteps total array size 0x190. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
ERROR 2019-04-25T06:28:56Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x1101 (Type = 8) | Mapping Memory from 0x0 to 0x1c0 oversteps total array size 0x190. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
ERROR 2019-04-25T06:28:56Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x1104 (Type = 8) | Mapping Memory from 0x0 to 0x1c0 oversteps total array size 0x190. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
ERROR 2019-04-25T06:28:56Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x1107 (Type = 8) | Mapping Memory from 0x0 to 0x1c0 oversteps total array size 0x190. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
ERROR 2019-04-25T06:28:56Z: gfx_backend_vulkan: [Validation]  [ VUID-vkMapMemory-size-00681 ] Object: 0x110a (Type = 8) | Mapping Memory from 0x0 to 0x1c0 oversteps total array size 0x190. The Vulkan spec states: If size is not equal to VK_WHOLE_SIZE, size must be less than or equal to the size of the memory minus offset (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkMapMemory-size-00681)
```

**Comment by kvark** [comment]
Thank you for filing the issue! It's addressed as a part of #127, which is on its way to landing.

**Comment by m4b** [comment]
That was fast!
